contracts/bridge/x/v4/RegistrationV4.sol

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Immutable Pty Ltd 2018 - 2024
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import {CoreV4} from "./CoreV4.sol";
+
+contract RegistrationV4 {
+    CoreV4 public immutable imx;
+
+    constructor(address payable _imx) {
+        imx = CoreV4(_imx);
+    }
+
+    function getVersion() external view returns (string memory) {
+        return imx.VERSION();
+    }
+
+    function isRegistered(uint256 starkKey) public view returns (bool) {
+        return imx.getEthKey(starkKey) != address(0);
+    }
+
+    function registerAndWithdrawAll(address ethKey, uint256 starkKey, bytes calldata signature, uint256 assetType)
+        external
+    {
+        if (!isRegistered(starkKey)) {
+            imx.registerEthAddress(ethKey, starkKey, signature);
+        }
+        withdrawAll(uint160(ethKey), starkKey, assetType);
+    }
+
+    function withdrawAll(uint256 ethKey, uint256 starkKey, uint256 assetType) public {
+        uint256 ethKeyBalance = imx.getWithdrawalBalance(ethKey, assetType);
+        uint256 starkKeyBalance = imx.getWithdrawalBalance(starkKey, assetType);
+        require(ethKeyBalance != 0 || starkKeyBalance != 0, "No funds to withdraw");
+
+        if (ethKeyBalance > 0) {
```

Comment:
Could any registered starkKey holder trigger a withdrawal for an ethKeyBalance?

Reply:
yes and that's not a risk like explained here https://github.com/immutable/contracts/pull/179/files/8bb68f46ae9ed3add50ae8a6b1159f475bb063c5#r1491969574. This would just mean that another wallet is trying to claim a withdrawal for a third party and in the end the third party will get their funds